### ISSUE TITLE
New edit script implementation?

### ISSUE THREAD

**Issue by kylebgorman** [CUNY-CL/udtube#9]
My edit script library is more or less copied over from a Twitter DM with Chrupala years ago. It might return different results than the [edit script implementation from the Strakaverse](https://github.com/Hyperparticle/udify/blob/master/udify/dataset_readers/lemma_edit.py) used in UDPipe 2 prototype and UDify. If it does, we probably should just substitute that one in.

**Comment by kylebgorman** [comment]
I took a quick look at what's there and compared to what we actually have. 

UDify uses a variant of Levenshtein distance (e.g., computed with the standard Wagner-Fischer algorithm) to find the edit script. I believe there is at least the possibility of "ties" in this situation. For instance, these are equally good alignments of "bana" to "banana" according to Levenshtein distance:

```
bana
====ii
banana

ba  na
==ii==
banana
```

It took me a really long time to figure out what UDify does, but suffice to say it turns to heuristics here. 

Our current implementation uses `difflib`'s `SequenceMatcher`. The algorithm this implements is also heuristic, but in a more principled way: it greedily finds the longest contiguous match, then recurses until there are no matches. The method is described [here](https://en.wikipedia.org/wiki/Gestalt_pattern_matching). The implementation is O(mn) just like UDify's. (It is incorrectly described as "quadratic" elsewhere, and the original algorithm was worse-case "cubic" whatever that means.) Arguably this does have a semi-principled way to resolve ties: it gives precedence to long match sequences, and thus it prefers the first alignment I gave above because it has the longest contiguous subsequence match `bana`. That was very much the intention of the designers of `difflib` and here it sort of matches my intuition about what the "best" alignment is. (This intuition is simply that infixation is rarer than suffixation.) Note the question of `EditScript` vs. `ReverseEditScript` is independent of the alignment subproblem.

So I am inclined to conclude that we already have the best implementation here. @DanielYakubov any thoughts?
